store/src/java/com/zimbra/cs/store/file/FileBlobStore.java

```diff
@@ -367,12 +367,17 @@ private boolean deleteFile(File file) throws IOException {
 
     @Override
     public MailboxBlob getMailboxBlob(Mailbox mbox, int itemId, int revision, String locator, boolean validate) throws ServiceException {
-        short volumeId = Short.valueOf(locator);
-        File file = getMailboxBlobFile(mbox, itemId, revision, volumeId, validate);
-        if (file == null) {
-            return null;
+        try {
+            short volumeId = Short.valueOf(locator);
```

Comment:
Check for `locator` is  a valid number or not and immediately throws an exception and stops further processing which will reduce the unnecessary execution of the remaining code. This will also avoids relying on exceptions for the control flow.

Reply:
It should be always fail fast.

Reply:
For now we cannot do modifications in this PR as the ticket is already closed on May 17th. We need to do again the Pre-UAT even for a single change which we cannot accommodate right now. We will be considering it next time